### PR TITLE
OpenAI REST API, Streaming, and Apollo 4

### PR DESCRIPTION
**Pull request by jmacAJ**
This PR is a lot of file changes because I changed the /graphql directory name to /server and that touched a lot of references.  For the code in the PR I implemented a new REST interface that's driven from pathway properties (in /server/rest.js).  Essentially you can have any pathway emulate an OpenAI model (or wildcard) and, if REST interfaces are turned on in Cortex, it will respond with OpenAI chat and completion APIs for that model.  This PR also updates the server dependencies to Apollo 4 and fixed up the internal streaming response implementation so streaming can be passed through to REST calls.

### CODE REVIEW

File: server/graphql.js [hunk 1]
```diff
@@ -111,13 +83,15 @@ const getTypedefs = (pathways) => {
...
 
+// Resolvers for GraphQL
 const getResolvers = (config, pathways) => {
     const resolverFunctions = {};
     for (const [name, pathway] of Object.entries(pathways)) {
         if (pathway.disabled) continue; //skip disabled pathways
         resolverFunctions[name] = (parent, args, contextValue, info) => {
             // add shared state to contextValue
             contextValue.pathway = pathway;
+            contextValue.config = config;
```
Comment (ademgcl) [inline server/graphql.js:94]: i think this usage of changing contextValue in resolver might be against the best practices but works

File: server/rest.js [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+                jsonResponse.id = `chatcmpl-${requestId}`;
+
+                res.json(jsonResponse);
+            }
+
+        });
+
+        app.get('/v1/models', async (req, res) => {
+            const openAIModels = { ...openAIChatModels, ...openAICompletionModels };
+            const defaultModelId = 'gpt-3.5-turbo';
```
Comment (ademgcl) [inline server/rest.js:293]: should we do sth about this default guy

File: server/graphql.js (hunk 1 above)
Comment (jmacAJ) [reply server/graphql.js:94]: Yeah it definitely can overwrite pathway and config unexpectedly, but in this case we always want to control what those are anyway so I'll allow it.

File: server/rest.js (hunk 2 above)
Comment (jmacAJ) [reply server/rest.js:293]: We have to have at least one model available.  I suppose we could make it a config parameter, but this seems clear enough.  You only see this code when there are no other specific handlers (e.g. only wildcard handlers).

**Pull request merged**